rust/lance-linalg/src/kernels.rs

```diff
@@ -130,6 +134,45 @@ pub fn normalize<T: Float + Sum>(v: &[T]) -> impl Iterator<Item = T> + '_ {
     v.iter().map(move |&x| x / l2_norm)
 }
 
+fn do_normalize_arrow<T: ArrowPrimitiveType>(arr: &dyn Array) -> Result<ArrayRef>
+where
+    <T as ArrowPrimitiveType>::Native: Float + Sum,
+{
+    let v = arr.as_primitive::<T>();
+    Ok(Arc::new(PrimitiveArray::<T>::from_iter_values(normalize(v.values()))) as ArrayRef)
```

Comment:
FYI I plan to fix this soon, but `normalize` doesn't dispatch to the optimized SIMD kernels we have yet.

Reply:
🔥